src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -573,3 +580,322 @@ def _is_poller(obj):
+                            would cause a loss of data. (bool)
+            - exception_handler: Exception handler for handling non-standard exceptions (function)
+            - no_wait_param: The name of a boolean parameter that will be exposed as `--no-wait` to skip long
+              running operation polling. (string)
+            - transform: Transform function for transforming the output of the command (function)
+            - table_transformer: Transform function or JMESPath query to be applied to table output to create a
+                                 better output format for tables. (function or string)
+            - resource_type: The ResourceType enum value to use with min or max API. (ResourceType)
+            - min_api: Minimum API version required for commands within the group (string)
+            - max_api: Maximum API version required for commands within the group (string)
+        :rtype: None
+        """
+        self._check_stale()
+        merged_kwargs = self.group_kwargs.copy()
+        custom_command_type = merged_kwargs.get('custom_command_type', None)
+        if not custom_command_type:
+            raise CLIError('Module does not have `custom_command_type` set!')
+        if custom_command_type:
+            merged_kwargs.update(custom_command_type.settings)
+        merged_kwargs.update(kwargs)
+
+        command_name = '{} {}'.format(self.group_name, name) if self.group_name else name
+        self.command_loader._cli_command(command_name,  # pylint: disable=protected-access
+                                         operation=merged_kwargs['operations_tmpl'].format(method_name),
+                                         **merged_kwargs)
+
+    # pylint: disable=no-self-use
+    def _resolve_operation(self, kwargs, name, command_type=None, source_kwarg='command_type'):
+
+        allowed_source_kwargs = ['command_type', 'custom_command_type']
+        if source_kwarg not in allowed_source_kwargs:
+            raise ValueError("command authoring error: 'source_kwarg' value '{}'. Allowed values: {}".format(
+                source_kwarg, ' '.join(allowed_source_kwargs)))
+
+        operations_tmpl = None
+        if command_type:
+            # Top priority: specified command_type for the parameter
+            operations_tmpl = command_type.settings.get('operations_tmpl', None)
+
+        if not operations_tmpl:
+            # Second source: general operations_tmpl set for the command kwargs
+            operations_tmpl = kwargs.get('operations_tmpl', None)
+
+        if not operations_tmpl:
+            # Final source: retrieve the operations_tmpl from the relevant 'command_type' or 'custom_command_type'
+            command_type = kwargs.get(source_kwarg, None)
+            operations_tmpl = command_type.settings.get('operations_tmpl', None)
+
+        if not operations_tmpl:
+            raise ValueError("command authoring error: unable to resolve 'operations_tmpl'")
```

Comment:
I'm thinking this could be improved with using `next(something_here, None)`.
Where `something_here` is a list/tuple. So the first one that returns a value is the one you use.

Not sure if it's cleaner though.